activerecord/lib/active_record/associations/collection_association.rb

```diff
@@ -119,7 +119,7 @@ def build(attributes = nil, &block)
       def concat(*records)
         records = records.flatten
         if owner.new_record?
-          load_target
+          loaded!
```

Comment:
We need to call `loaded!` here because [this test](https://github.com/rails/rails/blob/c6b227b95b631ead7ffd5c8984b16b0fbe5b3ca4/activerecord/test/cases/associations_test.rb#L161) expects it (ref https://github.com/rails/rails/commit/df0765d8dc013ccd399d994109eaf39be4c24e81).

But I don't think we to call `load_target`; as this is a new record, I don't expect it to find any records.